src/libasr/codegen/x86_assembler.cpp

```diff
@@ -10,6 +10,26 @@
 
 namespace LCompilers {
 
+void X86Assembler::save_binary64(const std::string &filename) {
+    Vec<uint8_t> header = create_elf64_x86_header(
+        m_al, origin(), get_defined_symbol("_start").value,
+        compute_seg_size("text_segment_start", "text_segment_end"),
+        compute_seg_size("data_segment_start", "data_segment_end"));
+    {
+        std::ofstream out;
+        out.open(filename);
+        out.write((const char*) header.p, header.size());
+        out.write((const char*) m_code.p, m_code.size());
+    }
+#ifdef LFORTRAN_LINUX
+    std::string mode = "0755";
+    int mod = strtol(mode.c_str(), 0, 8);
```

Comment:
I think this is the same as:
```suggestion
    int mod = 0755;
```